html/browsers/the-window-object/BarProp.window.js

```diff
@@ -0,0 +1,53 @@
+function assert_barProps(barPropObjects, visible) {
+  let lastBarProp = undefined;
+  for (let barProp of barPropObjects) {
+    const currentBarProp = barProp;
+    assert_not_equals(currentBarProp, lastBarProp, "BarBrop objects of different properties are identical");
+    assert_equals(currentBarProp.visible, visible, "a BarProp's visible is wrong");
+    lastBarProp = currentBarProp;
+  }
+}
+
+function assert_identical_barProps(barProps, w, oldBarPropObjects, visible) {
+  barProps.map(val => w[val]).map((val, index) => {
+    assert_equals(val, oldBarPropObjects[index], "BarProp identity not preserved");
+  });
+  assert_barProps(oldBarPropObjects, visible);
+}
+
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe")),
+        frameW = frame.contentWindow,
+        barProps = ["locationbar", "menubar", "personalbar", "scrollbars", "statusbar", "toolbar"],
+        barPropObjects = barProps.map(val => frameW[val]);
+
+  assert_barProps(barPropObjects, true);
```

Comment:
Honestly it's pretty surprising these count as visible for an iframe. Something for https://github.com/whatwg/html/issues/4431 I suppose.